statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,726 @@
+                   'add': np.add,
+                   None: lambda l,
+                   b: l}[trend]
+        detrend = {'mul': np.divide,
+                   'add': np.subtract,
+                   None: lambda l,
+                   b: 0}[trend]
+        dampen = {'mul': np.power,
+                  'add': np.multiply,
+                  None: lambda b,
+                  phi: 0}[trend]
+        if seasonal == 'mul':
+            for i in range(1, n + 1):
+                l[i] = y_alpha[i - 1] / s[i - 1] + \
+                    (alphac * trended(l[i - 1], dampen(b[i - 1], phi)))
+                if trending:
+                    b[i] = (beta * detrend(l[i], l[i - 1])) + \
+                        (betac * dampen(b[i - 1], phi))
+                s[i + m - 1] = y_gamma[i - 1] / \
+                    trended(l[i - 1], dampen(b[i - 1], phi)) + \
+                    (gammac * s[i - 1])
+            slope = b[:i].copy()
+            season = s[m:i + m].copy()
+            l[i:] = l[i]
+            b[:i] = dampen(b[:i], phi)
+            b[i:] = dampen(b[i], phi_h)
+            s[i + m - 1:] = [s[(i - 1) + j % m] for j in range(h + 1)]
+            fitted = trended(l, b) * s[:-m]
+        elif seasonal == 'add':
+            for i in range(1, n + 1):
+                l[i] = y_alpha[i - 1] - (alpha * s[i - 1]) + \
+                    (alphac * trended(l[i - 1], dampen(b[i - 1], phi)))
+                if trending:
+                    b[i] = (beta * detrend(l[i], l[i - 1])) + \
+                        (betac * dampen(b[i - 1], phi))
+                s[i + m - 1] = y_gamma[i - 1] - \
+                    (gamma * trended(l[i - 1],
+                                     dampen(b[i - 1], phi))) + (gammac * s[i - 1])
+            slope = b[:i].copy()
+            season = s[m:i + m].copy()
+            l[i:] = l[i]
+            b[:i] = dampen(b[:i], phi)
+            b[i:] = dampen(b[i], phi_h)
+            s[i + m - 1:] = [s[(i - 1) + j % m] for j in range(h + 1)]
+            fitted = trended(l, b) + s[:-m]
+        else:
+            for i in range(1, n + 1):
+                l[i] = y_alpha[i - 1] + \
+                    (alphac * trended(l[i - 1], dampen(b[i - 1], phi)))
+                if trending:
+                    b[i] = (beta * detrend(l[i], l[i - 1])) + \
+                        (betac * dampen(b[i - 1], phi))
+            slope = b[:i].copy()
+            season = s[m:i + m].copy()
+            l[i:] = l[i]
+            b[:i] = dampen(b[:i], phi)
+            b[i:] = dampen(b[i], phi_h)
+            fitted = trended(l, b)
+        level = l[:i].copy()
+        if use_boxcox or use_boxcox == 'log':
+            fitted = inv_boxcox(fitted, lamda)
+            # TODO: Does it make sense to have a inv_boxcox transform of the level and trend?
+            #level = inv_boxcox(level, lamda)
+            #slope = inv_boxcox(slope, lamda)
+            # TODO: How do we deal with the inv_boxcox transform on negative seasonal components?
```

Comment:
my guess here is that seasonal and other components need to be computed as the difference between inv_boxcox transformed sums.
e.g. `seasonal_effect = fitted - inv_boxcox(fitted_without_seasonal)`
(just a guess because I haven't gone through much of the PR/algorithm yet.)